### ISSUE TITLE
Get rid of NO_MORE_BYTECODES

### ISSUE THREAD

**Issue by rwy7** [b9org/b9#33]
We don't really need null terminated bytecode sequences. As part of getting b9 back to a smaller set of instructions, we want to remove this condition. Instead, programs should just return from the top level instruction.

**Comment by jduimovich** [comment]
That will work for the interpreter which never gets to the null test because the return instruction will just return.  The JIT however uses the length using computeNumberOfBytecodes so we'd need to add a length field to the bytecodes somewhere.

**Comment by youngar** [comment]
Yeah, we definitely need them for the `ByteCodeBuilder` table.  I transitioned the FunctionTable to use a count to get rid of `NO_MORE_FUNCTIONS` already.  It's a pretty similar change.  I'm going to be working on getting the JIT up and running in the `next` branch soon, and I will probably tackle this after that.  If anyone else want's to do it, let me know.

**Comment by rwy7** [comment]
As an update, we changed `NO_MORE_BYTECODES` to an `Instruction` constant, `END_SECTION`, and added a corresponding `ByteCode::END_SECTION`. Now the CPP define is gone, but the concept of a null terminator is still there.
